sensor/common/scan/scan.go

```diff
@@ -363,6 +363,9 @@ func (s *LocalScan) fetchSignatures(ctx context.Context, errorList *errorhelpers
 // scanImage will scan the given image and return its components.
 func scanImage(ctx context.Context, image *storage.Image,
 	registry registryTypes.ImageRegistry, scannerClient scannerclient.ScannerClient) (*scannerclient.ImageAnalysis, error) {
+	if registry == nil {
```

Comment:
Rather than add preconditions to a non-exported function I think it might be better to track down all call sites (I could only find `fetchImageAnalysis` without an IDE handy) and make sure passing a nil registry is not possible there.

Reply:
This was addressed in #7854